projects/mysadna2/src/app/data.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { catchError, map, tap } from 'rxjs/operators';
+
+import { User } from './user';
+import { Project } from './project';
+
+import {USERS} from './mock-users';
+import {PROJECTS} from './mock-projects';
+
+const httpOptions = {
+  headers: new HttpHeaders({ 'Content-Type': 'application.json' })
+};
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DataService {
+  //private usersUrl = 'api/users'; //For web api
+  //private projectsUrl = 'api/projects';
+
+  constructor() { }
+//private http: HttpClient
+  /**
+  * Handle Http operation that failed.
+  * Let the app continue.
+  * @param operation - name of the operation that failed
+  * @param result - optional value to return as the observable result
+  */
+  private handleError<T> (operation = 'operation', result?: T) {
+    return (error: any): Observable<T> => {
+
+      // TODO: send the error to remote logging infrastructure
+      console.error(error); // log to console instead
+
+      // TODO: better job of transforming error for user consumption
+      console.log(`${operation} failed: ${error.message}`);
+
+      // Let the app keep running by returning an empty result.
+      return of(result as T);
+    };
+  }
+
+  /** GET users from the server (currently local const USERS)*/
+  getAllUsers(): Observable<User[]> {
+    return of(USERS);
+  }
+
+  /**GET projects from the server (currently local const PROJECTS)*/
```

Comment:
Why do you use multiline comments here?

Reply:
Once again, I was using a file from angular's tutorial as reference. For some reason, they used multiline comments for single line comments there, and I didn't think to change that.